src/java.base/share/classes/jdk/internal/module/ModuleLoaderMap.java

```diff
@@ -85,10 +84,10 @@ private static final class Mapper implements Function<String, ClassLoader> {
 
         @Override
         public ClassLoader apply(String name) {
-            Integer loader = map.get(name);
-            if (loader == APP_LOADER_INDEX) {
+            String loader = map.get(name);
+            if (APP_LOADER_NAME.equals(loader)) {
                 return APP_CLASSLOADER;
-            } else if (loader == PLATFORM_LOADER_INDEX) {
+            } else if (PLATFORM_LOADER_NAME.equals(loader)) {
                 return PLATFORM_CLASSLOADER;
             } else { // BOOT_LOADER_INDEX
```

Comment:
Drop the `// BOOT_LOADER_INDEX`?

Reply:
Done.

Reply:
Looks like the trivial update also needs to be reviewed, after the previous review/approval. @shipilev can you review again? Thanks